### PR TITLE
reference: update doc about expr_pushdown_blacklist

### PR DESCRIPTION
**Pull request by windtalker**
### What is changed, added or deleted? (Required)

TiDB `expr_pushdown_blacklist` is enhanced, so the document need to be updated.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- Other reference link(s): https://github.com/pingcap/tidb/pull/16389

### CODE REVIEW

**Comment by TomShawn** [comment]
@windtalker Please involve a technical review.

File: reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 1]
```diff
@@ -23,11 +23,27 @@ category: reference
...
+```
+
+其中，`name` 为禁止下推的函数名，`store_type` 指明希望禁止该函数下推到哪个 store，目前 TiDB 支持三种 store，分别为 `tikv`， `tidb` 和 `tiflash`，`store_type` 不区分大小写，如果需要禁止向多个 store 下推的话各个 store 之间以逗号隔开，`reason` 列可以记录该函数被加入黑名单的原因。
+
 ### 加入黑名单
 
 执行以下步骤，可将一个或多个函数或运算符加入黑名单：
 
-1. 向 `mysql.expr_pushdown_blacklist` 插入对应的函数名或运算符名。
+1. 向 `mysql.expr_pushdown_blacklist` 插入对应的函数名或运算符名以及希望禁止下推的 store 名。
```
Comment (lzmhhh123) [inline reference/sql/functions-and-operators/expressions-pushed-down.md:46]:
```suggestion
1. 向 `mysql.expr_pushdown_blacklist` 插入对应的函数名或运算符名以及希望禁止下推的存储类型集合。
```

File: reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 2]
```diff
@@ -23,11 +23,27 @@ category: reference
...
+| Field      | Type         | Null | Key  | Default           | Extra |
++------------+--------------+------+------+-------------------+-------+
+| name       | char(100)    | NO   |      | NULL              |       |
+| store_type | char(100)    | NO   |      | tikv,tiflash,tidb |       |
+| reason     | varchar(200) | YES  |      | NULL              |       |
++------------+--------------+------+------+-------------------+-------+
+3 rows in set (0.00 sec)
+```
+
+其中，`name` 为禁止下推的函数名，`store_type` 指明希望禁止该函数下推到哪个 store，目前 TiDB 支持三种 store，分别为 `tikv`， `tidb` 和 `tiflash`，`store_type` 不区分大小写，如果需要禁止向多个 store 下推的话各个 store 之间以逗号隔开，`reason` 列可以记录该函数被加入黑名单的原因。
```
Comment (lzmhhh123) [inline reference/sql/functions-and-operators/expressions-pushed-down.md:40]:
```suggestion
其中，`name` 为禁止下推的函数名，`store_type` 指明希望禁止该函数下推到哪些存储引擎，目前 TiDB 支持三种存储引擎，分别为 `tikv`， `tidb` 和 `tiflash`，`store_type` 不区分大小写，如果需要禁止向多个存储引擎下推的话各个存储之间以逗号隔开，`reason` 列可以记录该函数被加入黑名单的原因。
```

**Comment by lzmhhh123** [comment]
Should be cherry-picked to release-3.1?

**Comment by windtalker** [comment]
> Should be cherry-picked to release-3.1?

No, the change of `expr_pushdown_blacklist` will not be cherry picked to release-3.1

File: reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 3]
```diff
@@ -23,11 +23,27 @@ category: reference
...
+| Field      | Type         | Null | Key  | Default           | Extra |
++------------+--------------+------+------+-------------------+-------+
+| name       | char(100)    | NO   |      | NULL              |       |
+| store_type | char(100)    | NO   |      | tikv,tiflash,tidb |       |
+| reason     | varchar(200) | YES  |      | NULL              |       |
++------------+--------------+------+------+-------------------+-------+
+3 rows in set (0.00 sec)
+```
+
+其中，`name` 为禁止下推的函数名，`store_type` 指明希望禁止该函数下推到哪些存储引擎，目前 TiDB 支持三种存储引擎，分别为 `tikv`， `tidb` 和 `tiflash`，`store_type` 不区分大小写，如果需要禁止向多个存储引擎下推的话各个存储之间以逗号隔开，`reason` 列可以记录该函数被加入黑名单的原因。
```
Comment (TomShawn) [inline reference/sql/functions-and-operators/expressions-pushed-down.md:40]:
```suggestion
以上结果字段解释如下：

+ `name` 为禁止下推的函数名。
+ `store_type` 指明希望禁止该函数下推到哪些存储引擎。目前 TiDB 支持三种存储引擎，分别为 `tikv`、`tidb` 和 `tiflash`。`store_type` 不区分大小写，如果需要禁止向多个存储引擎下推，各个存储之间应以逗号隔开。
+ `reason` 列可以记录该函数被加入黑名单的原因。
```

**Pull request merged**